wgpu/src/lib.rs

```diff
@@ -3355,6 +3367,40 @@ impl<'a> RenderBundleEncoder<'a> {
     }
 }
 
+/// A write-only view into a staging buffer
+pub struct QueueWriteBufferView<'a> {
+    queue: &'a Queue,
+    buffer: &'a Buffer,
+    offset: BufferAddress,
+    inner: QueueWriteBuffer,
+}
+
+impl<'a> std::ops::Deref for QueueWriteBufferView<'a> {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
+        panic!("QueueWriteBufferView is write-only!");
```

Comment:
I don't think you can. See https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8507973a9c6ed94ee89eb6670980314f

As far as I understand, rust reborrows through the mutable reference which in turn panics in this case.

`BufferViewMut` also has the same implementation https://github.com/gfx-rs/wgpu/blob/0d4d3f4ceb0c07de38064533c3409a2f8c1d8d95/wgpu/src/lib.rs#L2297-L2308

Reply:
You can still do the following casts: `&mut QueueWriteBufferView` -> `&mut [u8]` -> `&[u8]. With your example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d473956694eea896d1e8de9aa5ab9273.

Anyway, I guess it is fine to do this as it may prevent errors, as long as the buffer is properly initialized to prevent UB if someone reads from it anyway ; especially if the same pattern is used elsewhere in `wgpu`.

Reply:
That's only there to try and discourage reading - it will work and it's not UB - but reading from a write combined buffer is just comically slow.